include/lager.hrl

```diff
@@ -20,6 +20,48 @@
 -define(DEFAULT_SINK, lager_event).
 -define(ERROR_LOGGER_SINK, error_logger_lager_event).
 
+-define(METADATA(Extras), [{severity, info},
+                           {pid, self()},
+                           {node, node()},
+                           {module, ?MODULE},
+                           {function, ?FUNCTION_NAME},
+                           {function_arity, ?FUNCTION_ARITY},
+                           {file, ?FILE},
+                           {line, ?LINE} | Extras]).
+
+-define(log(Level, Format, Args, Safety),
+        lager:dispatch_log(?DEFAULT_SINK, Level, ?METADATA([]), Format, Args,
```

Comment:
This macro construction means that lager's parse transform will not do the "whether to log this message" evaluation at the code site.  I am mild to the need for a macro based interface (I personally would not use it) but I understand people may want that kind of facility.

A better construction would be something like:
```erlang
-define(log(Level, Format, Args), lager:Level(Format, Args)).
-define(log_debug(Format, Args), ?log(debug, Format, Args)).
```

The `lager:Level` construction *will* get transformed by lager's parse transformation pass, and that should normally be the way that log messages get dispatched.

Reply:
I actually meant to include the 'whether to log this message' code in the macros body. I can add that.

And the point is to not have to have the parse transform pass, so replacing it with the `lager` call would defeat the purpose :)

Reply:
Alright. You'll want to have a look at the `lager_transform` module injects then and use the macro to inline it.